runtime/src/system_instruction_processor.rs

```diff
@@ -108,7 +108,7 @@ fn allocate(
         return Err(SystemError::InvalidAccountDataLength.into());
     }
 
-    account.set_data(&vec![0; space as usize])?;
+    account.set_data(vec![0; space as usize])?;
```

Comment:
Again, `set_data_len(0)` followed by `set_data_len(space)` ?

Reply:
Why do you think that would be better here?

Reply:
Less assumptions, closer to intend. Let the runtime decide how it is implemented instead of forcing it to be a heap allocation.

Reply:
I don't know what `space` can be here, but `set_data_len(n)` is malloc + memset (one invariant we have is that when this is called, the account is empty, so set_data_len will always allocate), while `set_data(vec![0; space])` is only `calloc` so the latter seems better to me.

That said we do have that issue with jemalloc and TLB flushes, so if `space` is small, memset is actually going to perform better.

Reply:
> That said we do have that issue with jemalloc and TLB flushes, so if space is small, memset is actually going to perform better.

I think I'll make it `set_data_len()` conservatively until we fix the jemalloc issue, as I don't want to risk making it worse.